store/metrics.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"github.com/coreos/etcd/Godeps/_workspace/src/github.com/prometheus/client_golang/prometheus"
+)
+
+// Set of raw Prometheus metrics.
+// Labels
+// * action = declared in event.go
+// * outcome = Outcome
+// Do not increment directly, use Report* methods.
```

Comment:
In this particular case I'm sticking to the conventions that exist in `etcd` codebase, a separate `metrics.go` file.

I actually find that significantly easier to read. For Java code I tend to follow a reporter pattern :  separate object containing all of the monitoring and event-based logging for a given set of responsibilities (e.g. request handling), with an interface consisting of methods that represent events that happened in the actual logic of the service. It prevents littering the logic code with random counter increments and makes it easier to test.

I used similar `ReportIncomingRequest` methods here, as IMHO, it makes the code easier to read.